Drafts Classify text page for NLP docs (#1911)
elastic/stack-docs: docs/en/stack/ml/nlp/ml-nlp-classify-text.asciidoc

[[ml-nlp-classify-text]]
= Classify text

:keywords: {ml-init}, {stack}, {nlp}, {lang-ident}, text classification, \
zero-shot text classification

> :description: NLP tasks that classify input text into categories or determine \

Reviewer: ```suggestion
:description: NLP tasks that classify input text or determine \
```

the language of text. 

> These NLP tasks enable you to identify the language of text and categorize 
> unstructured input text:

Reviewer: ```suggestion
These NLP tasks enable you to identify the language of text and classify or label 
unstructured input text:
```

* <<ml-nlp-lang-ident>>
* <<ml-nlp-text-classification>>
* <<ml-nlp-zero-shot>>

include::ml-nlp-lang-ident.asciidoc[]

[discrete]
[[ml-nlp-text-classification]]
== Text classification

> Text classification assigns the input text to one of multiple classes or 
> categories that best describe the text. The categories used depend 

Reviewer: ```suggestion
Text classification assigns the input text to one of multiple classes
 that best describe the text. The classes used depend 
```
maybe "labels"? instead of "classes"?

and determine whether the following text is a news topic related to "SPORTS", 
"BUSINESS", "LOCAL", or "ENTERTAINMENT":

[source,js]
----------------------------------
...
{
    "input_text": "The Blue Jays played their final game in Toronto last night and came out with a win over the Yankees, highlighting just how far the team has come this season."
}
...
----------------------------------
// NOTCONSOLE

[discrete]
[[ml-nlp-zero-shot]]
== Zero-shot text classification

The zero-shot classification task offers the ability to classify text without 
training a model on a specific set of classes. Instead, you provide the classes 
when you deploy the model or at inference time. It uses a model trained on a 
large data set that has gained a general language understanding and asks the 
model how well the labels you provided fit with your text.

This task enables you to analyze and classify your input text even when you 
don't have sufficient training data to train a text classification model.

For example, you might want to perform multi-class classification and determine 
whether a news topic is related to "SPORTS", "BUSINESS", "LOCAL", or 

> "ENTERTAINMENT". However, in this case the model is not trained specifically for 
> news categorization; instead, the possible labels are provided together with the 
> input text at inference time:

Reviewer: ```suggestion
"ENTERTAINMENT". However, in this case the model is not trained specifically for 
news classification; instead, the possible labels are provided together with the 
input text at inference time:
```
